opm/simulators/flow/BlackoilModelNldd.hpp

```diff
@@ -459,6 +470,18 @@ class BlackoilModelNldd {
             const double tt2 = detailTimer.stop();
             report.assemble_time += tt2;
             report.assemble_time_well += tt2;
+
+            // Check if we should dampen. Only do so if wells are converged.
+            if (!convreport.converged() && !convreport.wellFailed()) {
+                bool oscillate = false;
+                bool stagnate = false;
+                const int numPhases = convergence_history.front().size();
+                detail::detectOscillations(convergence_history, iter, numPhases, 0.2, 1, oscillate, stagnate);
+                if (oscillate) {
+                    damping_factor *= 0.85;
```

Comment:
Is the magic number `0.85` based on some kind of empirical evidence?

Reply:
Not very, it was chosen to avoid going immediately to very harsh damping. I experimented with 0.8 first but did not observe significant differences. It might be a good idea to research this further, in followup work.